src/datasources/asset/AssetDataSource.ts

```diff
@@ -70,10 +96,203 @@ export class AssetDataSource extends DataSourceBase<AssetMetadataQuery> {
     return result;
   }
 
-  shouldRunQuery(_: AssetMetadataQuery): boolean {
+  async processUtilizationQuery(query: AssetUtilizationQuery, options: DataQueryRequest) {
+    const result: DataFrameDTO = { refId: query.refId, fields: [] };
+    let workspaceId = this.templateSrv.replace(query.workspace);
+    // fetch and process utilization raw data for chosen assets/systems
+    const utilization_array = await this.fetchAndProcessUtilizationData(query, options);
+    if (utilization_array.length) {
+      let utilization_array_with_alias: Array<{ id: string, datetimes: number[], values: number[], alias: string }>;
+      const idsArray = utilization_array.map((data) => {
+        return data.id
+      });
+      // find and add aliases to data
+      if (query.entityType === EntityType.System) {
+        const filterArr: string[] = [];
+        idsArray.forEach((id) => {
+          filterArr.push(`id == "${id}" `);
+        });
+        const filterStr = workspaceId ? `(${filterArr.join(' or ')}) and workspace = "${workspaceId}"` : filterArr.join(' or ');
+        const systems = await this.querySystems(filterStr, defaultProjection);
+        utilization_array_with_alias = utilization_array.map(system_utilization_df => {
+          const foundItem = systems.find((system: SystemMetadata) => system.id === system_utilization_df.id);
+          return { ...system_utilization_df, alias: foundItem?.alias ? foundItem.alias : 'System' };
+        });
+      } else {
+        const filterArr: string[] = [];
+        idsArray.forEach((id) => {
+          filterArr.push(`${AssetFilterProperties.AssetIdentifier} == "${id}"`);
+        });
+        const filterStr = workspaceId ? ` (${filterArr.join(' or ')}) and workspace = "${workspaceId}"` : filterArr.join(' or ');
+        const assetsResponse = await this.queryAssets(filterStr, -1);
+        utilization_array_with_alias = utilization_array.map(asset_utilization_df => {
+          const foundItem = assetsResponse.find((asset: AssetModel) => asset.id === asset_utilization_df.id);
+          return { ...asset_utilization_df, alias: foundItem ? foundItem.name : 'Asset' };
+        });
+      }
+      result.fields = [
+        { name: 'time', values: utilization_array_with_alias[0].datetimes },
+      ];
+      utilization_array_with_alias.forEach((data) => {
+        result.fields.push(
+          {
+            name: data.id,
+            values: data.values,
+            // default configuration for visualizations
+            config: {
+              displayName: data.alias,
+              unit: '%',
+              min: 0,
+              max: 100
+            }
+          }
+        );
+      })
+
+      return result;
```

Comment:
You can just return once at the end of the method and avoid using else branch with return.